frontend/packages/ceph-storage-plugin/integration-tests/utils/helpers.ts

```diff
@@ -100,3 +100,45 @@ export const isPodPresent = (pods, podName) => {
 export const getOSDPreparePodsCnt = (pods) =>
   pods.items.filter((pod) => getPodName(pod).includes(POD_NAME_PATTERNS.ROOK_CEPH_OSD_PREPARE))
     .length;
+
+export const getZoneIds = (nodes) =>
+  nodes.filter((node) => node.type === ZONE).map((node) => node.id);
+
+export const getNodeIds = (nodes) =>
+  nodes.filter((node) => node.type === HOST).map((node) => node.id);
+
+export const getOSDIds = (nodes) =>
+  nodes.filter((node) => node.type === OSD).map((node) => node.id);
+
+export const getNewOSDIds = (nodes, osds) =>
+  nodes.filter((node) => node.type === OSD && osds.indexOf(node.id) === -1).map((node) => node.id);
+
+// created dictionary for faster acess O(1)
+export const createOSDTreeMap = (nodes) => {
+  const tree = {};
+  nodes.forEach((node) => {
+    tree[node.id] = node;
+  });
+  return tree;
+};
+
+export const verifyZoneOSDMapping = (zones, osds, osdtree) => {
+  let filteredOsds = [];
+  zones.forEach((zone) => {
+    const hostId = osdtree[zone].children[0];
+    const len = osdtree[hostId].children.length;
```

Comment:
Won't this fail , if `tree` is an empty object  ?
Or its always be , if not, please guard for emptiness.

Reply:
It will not be empty object. So no need to guard.